File: est/toff/Task3v2.cpp
#include <iostream>
#include <sstream>
#include <vector>

int main() {
    std::string line;
    std::getline(std::cin, line);
    std::istringstream iss(line);
    std::vector<int> lineNumbers;
    int n;
    while (iss >> n) {
        lineNumbers.push_back(n);
    }

    int count = lineNumbers[0];
    int time = lineNumbers[1];

    std::getline(std::cin, line);
    std::istringstream iss2(line);
    std::vector<int> lineNumbers2;
    while (iss2 >> n) {
        lineNumbers2.push_back(n);
    }

    int min = lineNumbers2[0];
    int max = lineNumbers2[lineNumbers2.size() - 1];

    int employeeOut;
    std::cin >> employeeOut;

    if (count <= 0) return 0;

    if (employeeOut - min <= time || max - employeeOut <= time)
        std::cout << max - min << std::endl;
    else {
        int time1 = max - min + max - employeeOut;
        int time2 = max - min + employeeOut - min;
        std::cout << (time1 > time2 ? time2 : time1) << std::endl;
    }

    return 0;
}

File: est/toff/Task3v2.java
import java.util.Scanner;

public class Task3v2 {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        String[] lineNumbers = scan.nextLine().split(" ");
        int count = Integer.parseInt(lineNumbers[0]);
        int time = Integer.parseInt(lineNumbers[1]);

        lineNumbers = scan.nextLine().split(" ");
        int min = Integer.parseInt(lineNumbers[0]);
        int max = Integer.parseInt(lineNumbers[lineNumbers.length - 1]);

        int employeeOut = Integer.parseInt(lineNumbers[scan.nextInt() - 1]);
        scan.close();
        if (count <= 0) return;

        if (employeeOut - min <= time || max - employeeOut <= time)
            System.out.println(max - min);
        else {
            int time1 = max - min + max - employeeOut;
            int time2 = max - min + employeeOut - min;
            System.out.print(time1 > time2 ? time2 : time1);
        }
    }
}